test/v2/v2_test.go

```diff
@@ -139,196 +141,177 @@ func skipInCI(t *testing.T) {
+	err := testBasicDispersal(t, rand, paddedPayload, []core.QuorumID{0, 1})
 	require.NoError(t, err)
 }
 
 // Disperse a medium payload (between 100KB and 200KB).
 func TestMediumBlobDispersal(t *testing.T) {
-	skipInCI(t)
 	rand := random.NewTestRandom(t)
 	dataLength := 1024 * (100 + rand.Intn(100))
 	payload := rand.Bytes(dataLength)
 	paddedPayload := codec.ConvertByPaddingEmptyByte(payload)
-	require.Equal(t, calculateExpectedPaddedSize(dataLength), len(paddedPayload))
-	err := testBasicDispersal(t, rand, paddedPayload, len(paddedPayload), []core.QuorumID{0, 1})
+	err := testBasicDispersal(t, rand, paddedPayload, []core.QuorumID{0, 1})
 	require.NoError(t, err)
 }
 
 // Disperse a medium payload (between 1MB and 2MB).
 func TestLargeBlobDispersal(t *testing.T) {
-	skipInCI(t)
 	rand := random.NewTestRandom(t)
-	dataLength := int(1024 * 1024 * (1 + rand.Float64()))
+	dataLength := int(rand.Uint64n(targetConfig.MaxBlobSize/2) + targetConfig.MaxBlobSize/4)
 	payload := rand.Bytes(dataLength)
 	paddedPayload := codec.ConvertByPaddingEmptyByte(payload)
-	require.Equal(t, calculateExpectedPaddedSize(dataLength), len(paddedPayload))
-	err := testBasicDispersal(t, rand, paddedPayload, len(paddedPayload), []core.QuorumID{0, 1})
+	err := testBasicDispersal(t, rand, paddedPayload, []core.QuorumID{0, 1})
 	require.NoError(t, err)
 }
 
 // Disperse a small payload (between 1KB and 2KB) with a single quorum
 func TestSmallBlobDispersalSingleQuorum(t *testing.T) {
-	skipInCI(t)
 	rand := random.NewTestRandom(t)
 	desiredDataLength := 1024 + rand.Intn(1024)
 	payload := rand.Bytes(desiredDataLength)
 	paddedPayload := codec.ConvertByPaddingEmptyByte(payload)
-	require.Equal(t, calculateExpectedPaddedSize(desiredDataLength), len(paddedPayload))
-	err := testBasicDispersal(t, rand, paddedPayload, len(paddedPayload), []core.QuorumID{0})
+	err := testBasicDispersal(t, rand, paddedPayload, []core.QuorumID{0})
 	require.NoError(t, err)
 }
 
-// TODO:(dmanc): This test is failing. "Timed out waiting for blob to be confirmed"
 // Disperse a blob that is exactly at the maximum size after padding (16MB)
 func TestMaximumSizedBlobDispersal(t *testing.T) {
-	skipInCI(t)
-
-	t.Skipf("2mb is the max size in preprod") // TODO
-
 	rand := random.NewTestRandom(t)
-	originalSize, err := calculateOriginalSize(16 * 1024 * 1024)
-	require.NoError(t, err)
-	payload := rand.Bytes(originalSize)
-	padded := codec.ConvertByPaddingEmptyByte(payload)
-	lengthPadded := len(padded)
-	fmt.Printf("length: %d, originalSize: %d, lengthPadded: %d\n", len(payload), originalSize, lengthPadded)
-	require.Equal(t, calculateExpectedPaddedSize(originalSize), lengthPadded)
-	err = testBasicDispersal(t, rand, padded, lengthPadded, []core.QuorumID{0, 1})
+	dataLength := int(targetConfig.MaxBlobSize)
+	payload := rand.Bytes(dataLength)
+	paddedPayload := codec.ConvertByPaddingEmptyByte(payload)[:dataLength]
+
+	//require.Equal(t, calculateExpectedPaddedSize(dataLength), len(paddedPayload))
```

Comment:
commented out code intentionally left? (also elsewhere in this file)

Reply:
should be removed, is now removed